### PR TITLE
Include the wallet name in log messages relating to wallets

### PR DESCRIPTION
**Pull request by dooglus**
For #11317.

### CODE REVIEW

**Comment by promag** [comment]
How about something along one of these:
- `WalletLogPrintf(pwallet, ...);`
- `pwallet->LogPrintf(...);`

**Comment by jnewbery** [comment]
Tested ACK 9d68be4d868b31456bc22cb005c12ed650ca3ac0

**Comment by TheBlueMatt** [comment]
Do you want to include the wallet in the SELECTIONS debug prints? I dont really care either way cause it should be obvious, but for completeness figured I'd mention it.

utACK 9d68be4d868b31456bc22cb005c12ed650ca3ac0. I don't hugely care about having a WalletLogPrintf function/macro, its not all that different, but if you find it cleaner, do it.

**Comment by dooglus** [comment]
I added the wallet name to the SELECTCOINS message, and in four other places:

    -        LogPrint(BCLog::RPC, "Bumping fee and discarding dust output\n");
    +        LogPrint(BCLog::RPC, "[%s] Bumping fee and discarding dust output\n", pWallet->GetName());

    -            LogPrintf("Error getting wallet database cursor\n");
    +            LogPrintf("[%s] Error getting wallet database cursor\n", pwallet->GetName());

    -                LogPrintf("Error reading next record from wallet database\n");
    +                LogPrintf("[%s] Error reading next record from wallet database\n", pwallet->GetName());

    -                LogPrintf("%s\n", strErr);
    +                LogPrintf("[%s] %s\n", pwallet->GetName(), strErr);

then rebased.

File: src/wallet/db.cpp [hunk 1]
```diff
@@ -229,7 +229,7 @@ bool CDB::Recover(const std::string& filename, void *callbackDataIn, bool (*reco
 bool CDB::VerifyEnvironment(const std::string& walletFile, const fs::path& dataDir, std::string& errorStr)
 {
     LogPrintf("Using BerkeleyDB version %s\n", DbEnv::version(0, 0, 0));
```
Comment (jonasschnelli) [inline src/wallet/db.cpp]: Not really related, but this gets logged by `-wallet` instance?
Maybe a better approach would be to do a single `"Using wallets [x,y,z]\n"` would be better?

**Comment by jonasschnelli** [comment]
utACK f197a4c515757262eda0accbe762e311335be11b
I think we should be careful with adding more direct access to `pWallet`since we are planing to add dynamic loading/unloading.

**Comment by laanwj** [comment]
Concept ACK.

> I think we should be careful with adding more direct access to pWalletsince we are planing to add dynamic loading/unloading.

Agree here.

**Comment by luke-jr** [comment]
Concept ACK. I like `pwallet->LogPrintf(...);`

**Comment by sipa** [comment]
Concept ACK. I doubt this will interfere much with #11403, but I don't mind rebasing if that's a concern.

**Comment by PierreRochard** [comment]
@dooglus this is a good PR with unanimous concept ACKs, do you mind if I rebase your commit in a new PR so that we can get it in for 0.17?

**Comment by dooglus** [comment]
@PierreRochard I don't mind at all. Please do so.

**Pull request closed without merging**